concent_api/concent_api/system_check.py

```diff
@@ -346,3 +375,27 @@ def check_concents_time_settings(app_configs=None, **kwargs):  # pylint: disable
             if not isinstance(getattr(settings, concent_setting), int) or getattr(settings, concent_setting) < 0:
                 settings_wrong_value.append(create_error_24_if_concent_time_settings_have_wrong_value(concent_setting))
     return settings_not_defined + settings_wrong_value
+
+
+@register()
+def check_custom_protocol_times(app_configs=None, **kwargs):  # pylint: disable=unused-argument
+    errors = []
+    if not hasattr(settings, 'CUSTOM_PROTOCOL_TIMES'):
+        return [create_error_31_custom_protocol_times_is_not_set()]
+    if not isinstance(settings.CUSTOM_PROTOCOL_TIMES, bool):
+        return [create_error_32_custom_protocol_times_has_wrong_value()]
+    if settings.CUSTOM_PROTOCOL_TIMES is False:
+        for concent_setting_name, is_equal in {
+            'CONCENT_MESSAGING_TIME':   settings.CONCENT_MESSAGING_TIME == int(constants.CMT.total_seconds()),
+            'FORCE_ACCEPTANCE_TIME':    settings.FORCE_ACCEPTANCE_TIME  == int(constants.FAT.total_seconds()),
+            'PAYMENT_DUE_TIME':         settings.PAYMENT_DUE_TIME       == int(constants.PDT.total_seconds()),
+            'DOWNLOAD_LEADIN_TIME':     settings.DOWNLOAD_LEADIN_TIME   == int(constants.DOWNLOAD_LEADIN_TIME.total_seconds()),
+            'MINIMUM_UPLOAD_RATE':      settings.MINIMUM_UPLOAD_RATE    == constants.DEFAULT_UPLOAD_RATE,
+        }.items():
+            if not is_equal:
+                return [
```

Comment:
Maybe it would be better to iterate over all settings (and perhaps append errors to the list) than just returning on the first; that way we would have more information.

Reply:
:+1:

Reply:
I think it would be even better to just compare the whole dicts. Then, if they don't match, the Django prints a diff and you see exactly which values do not match.